prow/pod-utils/gcs/upload.go

```diff
@@ -189,12 +189,20 @@ func (w *openerObjectWriter) Write(p []byte) (n int, err error) {
 }
 
 func (w *openerObjectWriter) Close() error {
-	if w.writeCloser != nil {
-		err := w.writeCloser.Close()
-		w.writeCloser = nil
-		return err
+	if w.writeCloser == nil {
+		// Always create a writer even if Write() was never called
+		// otherwise empty files are never created, because Write() is
+		// never called for them
+		var err error
```

Comment:
Could this be 

```go
if _, err := w.Write([]byte()); err != nil {
    return err
}
err := w.writeCloser.Close()
w.writeCloser = nil
return err
```

Reply:
Yup would be better to avoid code duplication

Reply:
I kept the if `writeCloser == nil`. If I use the exact same you suggested it would always write an empty byte array, even if `Write()` was already used before (which should be 99% of the cases)